### ISSUE TITLE
Rust Language syntax highlighting is slightly off

### ISSUE THREAD

**Issue by ifacodes** [nopjmp/vscode-fairyfloss#5]
With cursive fonts this means that only let gets the italics, but also the color scheme is all messed up.

![image](https://user-images.githubusercontent.com/6687517/97794610-adecee00-1bd2-11eb-8416-6bb79f84e2e8.png)

**Comment by nopjmp** [comment]
Yeah, the syntax is off for quite a few languages. I need to go back through and properly learn how to handle these situations. I was struggling to make this port at the beginning.

**Comment by nopjmp** [comment]
I'm enabled a setting and with the default rust highlighting it looks fine, I know there are a lot of cases where things are weird right now. I'm going to end up working on the scopes list which can get hard to get every single programming language correct unless I'm missing something obvious.

**Comment by ifacodes** [comment]
> I'm enabled a setting and with the default rust highlighting it looks fine, I know there are a lot of cases where things are weird right now. I'm going to end up working on the scopes list which can get hard to get every single programming language correct unless I'm missing something obvious.

You can use array's for the scope if you need to, and then use different language setting strings for the scope and apply the settings to them all at once. 

I'm gonna look up the theme stuff so I can try help on this issue. That way I learn something and I can be of help at the same time.

**Comment by ifacodes** [comment]
https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide

This page is important for rust, because as far as I am aware, most rust users in vs-code are either using the Rust extension, or the rust-analyzer extension, and I believe both use the semantic-highlighting feature. 

Tomorrow morning I'm going to experiment a bit with getting the highlighting to match up correctly, and I'll report back here.

ifa_ <3
